Assets/LeapMotion/Scripts/HandPool.cs

```diff
@@ -59,35 +93,62 @@ public class ModelGroup {
-      HandRepresentation handRep = null;
-      List<IHandModel> models = new List<IHandModel>();
-      foreach (ModelGroup group in ModelPool) {
-        for (int i = 0; i < group.modelList.Count; i++) {
-          IHandModel model = group.modelList[i];
-          bool isCorrectHandedness;
-          Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
-          isCorrectHandedness = model.Handedness == handChirality;
-          if (!EnforceHandedness || model.Handedness == Chirality.Either) {
-            isCorrectHandedness = true;
-          }
-          bool isCorrectModelType;
-          isCorrectModelType = model.HandModelType == modelType;
-          if (isCorrectModelType && isCorrectHandedness) {
-            group.modelList.RemoveAt(i);
-            //--i;
-            models.Add(model);
-            break;
+      Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
+      HandRepresentation handRep = new HandProxy(this, hand, handChirality, modelType);
+      for (int i = 0; i < ModelPool.Count; i++) {
+        ModelGroup group = ModelPool[i];
+        if (group.IsEnabled) {
+          IHandModel model = group.TryGetModel(handChirality, modelType);
+          if (model != null) {
+            handRep.AddModel(model);
+            modelToHandRepMapping.Add(model, handRep);
           }
         }
       }
-      handRep = new HandProxy(this, models, hand);
+      ActiveHandReps.Add(handRep);
       return handRep;
     }
-
-    public void ReturnToPool(IHandModel model){
+    public void EnableGroup(string groupName) {
+      for (int i = 0; i < ModelPool.Count; i++) {
+        if (ModelPool[i].GroupName == groupName) {
+          ModelGroup group = ModelPool[i];
+          for (int hp = 0; hp < ActiveHandReps.Count; hp++ ) {
+            HandRepresentation handRep = ActiveHandReps[hp];
+            IHandModel model = group.TryGetModel(handRep.RepChirality, handRep.RepType);
+            if (model != null) {
+              handRep.AddModel(model);
+              modelToHandRepMapping.Add(model, handRep);
+            }
+          }
+          group.IsEnabled = true;
+        }
+      }
+    }
+    public void DisableGroup( string groupName) {
+      for (int i = 0; i < ModelPool.Count; i++) {
+        ModelGroup group = null;
+        if (ModelPool[i].GroupName == groupName) {
+          group = ModelPool[i];
+          for (int m = 0; m < group.modelsCheckedOut.Count; m++) {
+            IHandModel model = group.modelsCheckedOut[m];
+            HandRepresentation handRep;
+            if (modelToHandRepMapping.TryGetValue(model, out handRep)) {
+              handRep.RemoveModel(model);
+              group.ReturnToGroup(model);
+              m--;
+            }
+          }
+          Debug.Assert(group.modelsCheckedOut.Count == 0, group.GroupName + "'s modelsCheckedOut List has not been cleared");
```

Comment:
You can use the UnityEngine.Assertions namespace for greater control over asserts.  They allow the user to be able to turn asserts off (where they are compiled out), allow them to be logged to console, or allow them to throw exceptions.